src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurenceSerializerDefinition.cs

```diff
@@ -15,58 +15,17 @@ class LongValueOccurrenceSerializerDefinition : SerializationDefinition
         }
     }
 
-    class DurationRawDataSerializer : IMessageSerializer
+    class DurationRawDataSerializer : RawMessageSerializer<LongValueOccurrences>, IMessageSerializer
     {
-        static readonly object[] NoMessages = new object[0];
-
-        public void Serialize(object message, Stream stream)
-        {
-            throw new NotImplementedException();
-        }
-
         public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
         {
-            var reader = new BinaryReader(stream);
-
-            var version = reader.ReadInt64();
-
-            if (version == 1)
-            {
-                var baseTicks = reader.ReadInt64();
-                var count = reader.ReadInt32();
-
-                if (count == 0)
+            return DeserializeRawMessage(stream,
+                (t, r, m) =>
```

Comment:
My OCD here, but can we give these better names than `t`, `r`, and `m`? Especially since the func's types aren't named.